build/dsc/templates/dsc_type.rb.erb

```diff
@@ -63,9 +87,9 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
   # IsMandatory:  <%= property.required? ? 'True' : 'False' %>
   # Values:       <%= property.values ? property.values : 'None' %>
   newparam(:dsc_<%= property.name.downcase %><% if property.array? %>, :array_matching => :all<% end %>) do
-<%    if property.description -%>
-    desc "<%= property.description %>"
-<%    end -%>
+    desc "<%= property.name %><% if property.description -%> - <%= property.description %><% end %>
+      Expects input to be a <%= property.type %>.
```

Comment:
`property.type` is more or less the MOF type, so it might not be what we want to document for Puppet users

I think if we want to document the Puppet type here, we will need to build some sort of conversion table based on https://github.com/puppetlabs/puppetlabs-dsc/blob/master/build/dsc/property.rb#L75-L105